### ISSUE TITLE
Import VSC settings from JSON files

### ISSUE THREAD

**Issue by schickling** [gitpod-io/gitpod#10702]
I'd like to be able to import my [VSC settings via JSON files I keep in my dotfiles repo](https://github.com/schickling/dotfiles/tree/main/VSCode).

Ideally there's an easy way how this can be accomplished e.g. via the `gp` CLI. It would be acceptable for my use cases to only one-way **import** my VSC settings but ignore settings changes happening in Gitpod VSC.

More context:

- https://github.com/gitpod-io/gitpod/issues/3733#issuecomment-1157140735

**Comment by axonasif** [comment]
You can try [this](https://github.com/axonasif/dotfiles-sh#vscodeadd_settings) if you wish to do from dotfiles.

**Comment by schickling** [comment]
Bump.

**Comment by akosyakov** [comment]
@schickling could you try following:

- create settings.json file in your dotfiles repo
- in dotfiles script move it to "/workspace/.vscode-remote/data/Machine/settings.json

Not sure how it is going to work on workspace restart. I would imagine we should do it only on first start. But generally yes I think we can merge settings from different sources (team/project) and put them as machine settings before starting VS Code Server on very first workspace start. cc @loujaybee @jeanp413 

It is also related to support settings in .gitpod.yml, i.e. vscode.settings property.

**Comment by schickling** [comment]
Just gave this a try but didn't seem to work unfortunately (unless symbolic links don't work):

https://github.com/schickling/dotfiles/blob/main/install.sh#L53-L54

Any idea?

**Comment by schickling** [comment]
😬

**Comment by Return180bpm** [comment]
Also interested in this. Sync isn't working for me currently and manually managing my extensions in Gitpod is a chore that makes me not want to use Gitpod in the first place.

**Comment by akosyakov** [comment]
> Sync isn't working for me currently

@Return180bpm generally it should work, it is used daily by many users internally. If it does not please create a new issue with description what is wrong with Setting Sync.
